File: YangMingShanDemo/ObjC/DemoListViewController.m
//
//  DemoListViewController.m
//  YangMingShanDemo
//
//  Copyright 2016 Yahoo Inc.
//  Licensed under the terms of the BSD license. Please see LICENSE file in the project root for terms.
//

#import "DemoListViewController.h"

static NSString * const CellIdentifier = @"reuseIdentifier";

@interface DemoListViewController ()

@property (nonatomic, strong) NSArray *listArray;

@end

@implementation DemoListViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    NSDictionary *photoPickerInfo = @{@"title": @"YMSPhotoPicker",
                                      @"description": @"Photo & Album picker",
                                      @"segueIdentifier": @"goToPhotoViewIdentifier"};
    self.listArray = @[photoPickerInfo];
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.listArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:CellIdentifier];
    }

    NSDictionary *cellInfo = [self.listArray objectAtIndex:indexPath.row];

    cell.textLabel.text = [cellInfo objectForKey:@"title"];
    cell.detailTextLabel.text = [cellInfo objectForKey:@"description"];

    return cell;
}

#pragma mark - UITableViewDelegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSDictionary *cellInfo = [self.listArray objectAtIndex:indexPath.row];

    [self performSegueWithIdentifier:[cellInfo objectForKey:@"segueIdentifier"] sender:self];
}

@end

File: YangMingShanDemo/Swift/DemoListViewController.swift
//
//  ViewController.swift
//  YangMingShanDemo-Swift
//
//  Copyright 2016 Yahoo Inc.
//  Licensed under the terms of the BSD license. Please see LICENSE file in the project root for terms.
//

import UIKit
import YangMingShan


class DemoListViewController: UITableViewController {

    let listArray = [
        [
            "title": "YMSPhotoPicker",
            "description": "Photo & Album picker",
            "segueIdentifier": "goToPhotoViewIdentifier"
        ]
    ]
    let cellIdentifier = "reuseIdentifier"


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - UITableViewDataSources

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.listArray.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath)

        let cellInfo = self.listArray[(indexPath as NSIndexPath).row]

        cell.textLabel?.text = cellInfo["title"]
        cell.detailTextLabel?.text = cellInfo["description"]
        
        return cell;
    }

    // MARK: - UITableViewDelegate

    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let cellInfo = self.listArray[(indexPath as NSIndexPath).row]
        self.performSegue(withIdentifier: cellInfo["segueIdentifier"]!, sender: self)
    }
}

